modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -213,6 +222,71 @@ boolean evaluatePolicies(V value) {
         return true;
     }
 
+    /**
+     * ConcatenatedIterables which combines cache iterables and supports remove() functionality as well if underlying
+     * iterator supports it.
+     * @param <K> Type of key.
+     */
+    class ConcatenatedIterables<K> implements Iterable<K> {
```

Comment:
This inner class should be `static`, right? It doesn't need to hold a reference back to its enclosing class.

Reply:
Yes this should be static.